js-packages/web-console/src/lib/components/profiler/ProfilerLayout.svelte

```diff
@@ -90,6 +94,29 @@
     }
   }
 
+  // Handle leaf node double-click to navigate to SQL source position
+  function handleLeafNodeDoubleClick(nodeId: string) {
```

Comment:
I would split this function into two: the callback that gets the source, 
and then calls a second one which navigates to the sources